src/vivarium_gates_nutrition_optimization_child/components/wasting.py

```diff
@@ -0,0 +1,857 @@
+        return exposure
+
+
+class StaticChildWasting(Risk):
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        risk :
+            the type and name of a risk, specified as "type.name". Type is singular.
+        """
+        self.risk = EntityString("risk_factor.child_wasting")
+        name = "static_child_wasting"
+        self.configuration_defaults = self._get_configuration_defaults()
+        self.exposure_distribution = self._get_exposure_distribution()
+        self._sub_components = [self.exposure_distribution]
+
+        self._randomness_stream_name = f"initial_{name}_propensity"
+        self.propensity_column_name = f"{name}_propensity"
+        self.propensity_pipeline_name = f"{name}.propensity"
+        self.exposure_pipeline_name = f"{name}.exposure"
+
+
+class WastingTreatment(Risk):
+    def __init__(self, treatment_type: str):
+        super().__init__(treatment_type)
+
+        self.previous_wasting_column = f"previous_{data_keys.WASTING.name}"
+        self.wasting_column = data_keys.WASTING.name
+
+        self.treated_state = self._get_treated_state()
+
+    ##########################
+    # Initialization methods #
+    ##########################
+
+    def _get_treated_state(self) -> str:
+        return self.risk.name.split("_treatment")[0]
+
+    #################
+    # Setup methods #
+    #################
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self._register_on_time_step_prepare_listener(builder)
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        return builder.population.get_view(
+            [self.propensity_column_name, self.previous_wasting_column, self.wasting_column]
+        )
+
+    def _register_on_time_step_prepare_listener(self, builder: Builder) -> None:
+        builder.event.register_listener("time_step__cleanup", self.on_time_step_cleanup)
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_time_step_cleanup(self, event: Event):
+        pop = self.population_view.get(event.index)
+        propensity = pop[self.propensity_column_name]
+        remitted_mask = (pop[self.previous_wasting_column] == self.treated_state) & pop[
+            self.wasting_column
+        ] != self.treated_state
+        propensity.loc[remitted_mask] = self.randomness.get_draw(remitted_mask.index)
+        self.population_view.update(propensity)
+
+
+class DynamicChildWastingModel(DiseaseModel):
+    def setup(self, builder):
+        """Perform this component's setup."""
+        super(DiseaseModel, self).setup(builder)
+
+        self.configuration_age_start = builder.configuration.population.age_start
+        self.configuration_age_end = builder.configuration.population.age_end
+
+        cause_specific_mortality_rate = self.load_cause_specific_mortality_rate_data(builder)
+        self.cause_specific_mortality_rate = builder.lookup.build_table(
```

Comment:
I actually already do pass in get_data_functions={"cause_specific_mortality_rate": lambda *_: 0} in the disease model. I guess it's extraneous to even define the pipeline modifier for it here, then

The risk-attributable diseases act for severe and moderate PEM separately, i.e.
 `RiskAttributableDisease('cause.moderate_protein_energy_malnutrition', 'risk_factor.child_wasting')`
 `RiskAttributableDisease('cause.severe_protein_energy_malnutrition', 'risk_factor.child_wasting')`

One thing to note is that we still do use the total PEM CSMR in the data functions that determine incidence rates of MAM and SAM, namely in `load_daily_mortality_probabilities` (ln 701). I thought that we don't need to disaggregate them there, but I'm bringing it up in case I am wrong.

Reply:
I checked it out and it looks correct to me.